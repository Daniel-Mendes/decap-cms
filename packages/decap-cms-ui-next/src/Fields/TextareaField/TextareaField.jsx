import React, { useState } from 'react';
import { useTranslate } from 'react-polyglot';
import { default as TextareaAutoSize } from 'react-textarea-autosize';
import styled from '@emotion/styled';

import Field from '../../Field';

const StyledTextarea = styled(TextareaAutoSize)`
  color: ${({ theme }) => theme.color.highEmphasis};
  background: none;
  border: none;
  outline: none;
  resize: none;
  overflow: hidden;
  width: calc(100% + 32px);
  font-family: inherit;
  font-size: ${({ title }) => (title ? '2rem' : '1rem')};
  font-weight: ${({ title }) => (title ? 'bold' : 'normal')};
  letter-spacing: ${({ title }) => (title ? '-0.5px' : '0')};
  line-height: 1.5rem;
  caret-color: ${({ theme, error }) =>
    error ? theme.color.danger['900'] : theme.color.primary['800']};
  margin: -2rem -1rem -1rem -1rem;
  padding: 2rem 1rem 1rem 1rem;
  ${({ clickable }) => (clickable ? `cursor: pointer;` : ``)} ::placeholder {
    color: ${({ theme }) => theme.color.disabled};
  }
`;

function TextareaField({
  name,
  label,
  status,
  placeholder,
  description,
  value,
  onChange,
  focus,
  filled,
  error,
  errors,
  minRows = 5,
  ...props
}) {
  const [inputFocus, setInputFocus] = useState();
  const t = useTranslate();

  return (
    <Field
      label={label}
      labelTarget={name}
      status={status}
      description={description}
      focus={focus || inputFocus}
      filled={filled}
      error={error}
      errors={errors}
      {...props}
    >
      <StyledTextarea
        id={name}
        value={value || ''}
        placeholder={
          placeholder
            ? placeholder
            : label
            ? t('editor.editorControl.field.placeholder', { fieldLabel: label.toLowerCase() })
            : ''
        }
        minRows={minRows}
        onChange={onChange}
        onFocus={() => setInputFocus(true)}
        onBlur={() => setInputFocus(false)}
        error={error}
      />
    </Field>
  );
}

export default TextareaField;
