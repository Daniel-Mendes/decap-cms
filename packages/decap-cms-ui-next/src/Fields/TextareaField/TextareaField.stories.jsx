import React from 'react';
import styled from '@emotion/styled';
import { useArgs } from '@storybook/preview-api';

import TextareaField from '.';

export default {
  title: 'Fields/TextareaField',
  component: TextareaField,
  argTypes: {
    errors: {
      control: 'object',
      if: {
        arg: 'error',
      },
    },
  },
  args: {
    name: 'textarea',
    label: 'Textarea Field Label',
    placeholder: 'Type something here',
    description: 'This is a description',
    status: 'Required',
    value: '',
    minRows: 5,
    inline: false,
    error: false,
    errors: [
      {
        message: 'Error message.',
      },
    ],
  },
};

const StyledTextareaField = styled(TextareaField)`
  width: 33vw;
`;

export function _TextareaField(args) {
  const [{ value }, updateArgs] = useArgs();

  function handleChange(e) {
    updateArgs({ value: e.target.value });
  }

  return <StyledTextareaField {...args} value={value} onChange={handleChange} />;
}
