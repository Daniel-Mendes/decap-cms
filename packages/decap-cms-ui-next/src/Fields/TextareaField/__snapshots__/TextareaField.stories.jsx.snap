// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields/TextareaField _TextareaField smoke-test 1`] = `"<div class="ete4snt0 css-1j1yhdh-FieldWrap-StyledTextareaField e1x0ipyd6"><div class="css-t9hsmw-FieldInside-FieldInside e1x0ipyd5"><label for="textarea" class="css-r9gw8d-Label-StyledLabel e1x0ipyd3">Textarea Field Label <div color="neutral" radius="large" class="css-6930ng-BadgeWrap-BadgeWrap epfx4ki3">Required</div></label><div class="css-dj548v-ChildrenWrap e1x0ipyd4"><textarea id="textarea" placeholder="Type something here" class="css-1fnrfp2-StyledTextarea e17sl9lk0" style="height: 168px !important;"></textarea></div><p class="css-nalsj0-StyledDescription e1x0ipyd1">This is a description</p></div></div>"`;
